src/ensembl/src/shared/hooks/useOutsideClick.tsx

```diff
@@ -58,6 +58,13 @@ export default function useOutsideClick<T extends HTMLElement>(
     });
 
     document.addEventListener('click', handleClickOutside);
-    return () => document.removeEventListener('click', handleClickOutside);
+    return () => {
+      document.removeEventListener('click', handleClickOutside);
+      refs.forEach((ref) => {
+        ref?.current?.removeEventListener('click', () => {
+          clickedInside = true;
+        });
```

Comment:
wow, wow, I don't think that's how it works. You have to asssign `() => { clickedInside = true; }` to a variable, and then pass it to removeEventListener; like document.removeEventListener is doing.

Every 

```
() => {
  clickedInside = true;
}
```
is a declaration of a new anonymous function; while you need to remove the exact same function you passed when you added a listener.

Reply:
Ah yes! That's done now. I wonder what happens with the previous approach?